### ISSUE TITLE
508-defect-3: Buttons should have visible and understandable text labels

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#25277]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a voice assistive tech user, I want to be able to see a visible text label so I can say a command that will successfully register.

## Details

Because the copy button uses an icon "🔗" as it's visible label:
- voice assistive tech users won't know how to activate it confidently
- folks who don't understand what the icon means may not use it (for a variety of reasons from cultural, generational, etc)
- `aria-label` [does not translate in many cases](https://adrianroselli.com/2019/11/aria-label-does-not-translate.html)

The current `aria-label` also may need improvement to be more specific about its purpose.

**Angela's feedback**: The copy link button is an unusual way of providing a means to share a URL. Though it will ultimately be much simpler than more commonly used methods, it may initially cause confusion for users who mistake it for a means of navigation. It would be helpful to have some explanation of its intended purpose until users become accustomed to it.
Modify the button to say "copy a sharable link" instead of "copy link."

Providing a visible text label should address these needs.

It also will allow us to [pursue a multi-function button](https://adrianroselli.com/2021/01/multi-function-button.html) instead of relying on a tooltip announcement which can be difficult to read when broken on certain responsive sizes.

<img width="550" alt="Screen Shot 2021-05-26 at 11 20 05 AM" src="https://user-images.githubusercontent.com/14154792/119686437-53d4a400-be14-11eb-983f-2fcbf69f3f16.png">

## Acceptance Criteria

- [ ] A visible text label is present

## Proposed Solution 

I'd recommend using a combination of an anchor _link_ and a copy link _button_ OR having two versions of the copy link button: one with the text label when not included in a heading and one without the text label when included in a heading.

### A future state alternative ( from my comment in #25275 )
An alternative, perhaps more robust way of approaching this in a future iteration may be to split the "copy link" button apart into an anchor _link_ and a copy link _button_. We can accommodate more folks by providing alternative ways of copying a link.

[I like this accessible example of an anchor link](https://amberwilson.co.uk/blog/are-your-anchor-links-accessible/#what-are-anchor-links-exactly%3F) which is absolutely positioned off to the left and then flexed on smaller devices.

<img width="771" alt="Screen Shot 2021-05-14 at 9 24 58 AM" src="https://user-images.githubusercontent.com/14154792/118281719-8937d400-b49b-11eb-901f-b919d0847512.png">
Note: couldn't find the original share link icon, so please forgive the emoji usage

This would also resolve the following issues:
- We wouldn't have to worry about spacing or text alignment with the heading text anymore since we could use text (#) with `sr-only` text or an `aria-label` instead of an icon nested in a button (which makes alignment so tricky)
- We can include text copy in the button so it's more clear to everyone and it can accommodate folks who use voice assistive technology to click buttons e.g. With a visible label, it will be clear what to read out for the AT to take action e.g. "Click share link button." In the current state, a link icon could be interpreted as, "Click link button" which may not activate the button if their best guess of what the icon is doesn't match the accessible name.
- We can beef up the size of the share link button without worrying about it causing alignment issues with the anchor link.
- Users wouldn't mistake the copy link button in the heading to behave similarly to anchor links they see elsewhere on the web
- Instead of providing pop up feedback about the link being copied, we could just update the already existing label within the copy share link button

**Comment by SKasimow** [comment]
Thanks @Josh! We'll need to review with @jenniferlee-dsva once Fareez clears his plate :)

**Comment by SKasimow** [comment]
Closing this since we are removing anchor tag link experiment affordances.
